javaslang/src/main/java/javaslang/collection/Map.java

```diff
@@ -674,6 +674,33 @@ default int length() {
     @Override
     Map<K, V> replace(Tuple2<K, V> currentElement, Tuple2<K, V> newElement);
 
+    /**
+     * Replaces the entry for the specified key only if it is currently mapped to some value.
+     *
+     * @param key the key of the element to be substituted.
+     * @param value the new value to be associated with the key
+     * @return a new map containing key mapped to value if key was contained before. The old map otherwise.
+     */
+    Map<K, V> replaceValue(K key, V value);
+
+    /**
+     * Replaces the entry for the specified key only if currently mapped to the specified value.
+     *
+     * @param key the key of the element to be substituted.
+     * @param oldValue the expected current value that the key is currently mapped to
+     * @param newValue the new value to be associated with the key
+     * @return a new map containing key mapped to newValue if key was contained before and oldValue matched. The old map otherwise.
+     */
+    Map<K, V> replace(K key, V oldValue, V newValue);
+
+    /**
+     * Replaces each entry's value with the result of invoking the given function on that entry until all entries have been processed or the function throws an exception.
+     *
+     * @param function function transforming key and current value to a new value
+     * @return a new map with the same keySet but transformed values.
+     */
+    Map<K, V> replaceAll(BiFunction<? super K, ? super V, ? extends V> function);
+
```

Comment:
Mmhh, today I did a `push --force` on a branch/PR that wasn't merged yet. I think something went wrong o_O.

Don't know how this can be repaired 🙈 However, the current master contains these lines already: https://github.com/javaslang/javaslang/blob/master/javaslang/src/main/java/javaslang/collection/Map.java#L677-L703

@paplorinc is there some git magic that can be applied here? E.g. rebasing / rewriting ...